mongodb-atlas/src/main/java/io/vantiq/atlasConnector/AtlasStorageMgr.java

```diff
@@ -0,0 +1,544 @@
+                        "either an ObjectId or the string representation of an ObjectId."));
+            }
+            doc.append("_id", id);
+        }
+        return collectionFromStorageName(storageName, collection -> collection.insertOne(doc))
+                .firstOrError()
+                .map(insertOne -> externalizeId(doc));
+    }
+
+    static Map<String, Object> externalizeId(Map<String, Object> obj) {
+        if (obj.containsKey("_id")) {
+            obj.put("_id", obj.get("_id").toString());
+        }
+        return obj;
+    }
+    
+    @Override
+    public Flowable<Map<String, Object>>
+    insertMany(String storageName, Map<String, Object> storageManagerReference,List<Map<String, Object>> values) {
+        if (values.isEmpty()) {
+            return Flowable.empty();
+        }
+        List<Document> docs = new ArrayList<>();
+        values.forEach(value -> {
+            docs.add(new Document(value).append("_id", new ObjectId()));
+        });
+
+        return collectionFromStorageName(storageName, collection -> {
+            InsertManyOptions insertOpts = new InsertManyOptions().ordered(false);
+            return collection.insertMany(docs, insertOpts);
+        }).flatMap(result -> Flowable.fromIterable(docs))
+          .map(AtlasStorageMgr::externalizeId);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Maybe<Map<String, Object>>
+    update(String storageName, Map<String, Object> storageManagerReference, Map<String, Object> values,
+           Map<String, Object> qual) {
+        
+        Map<String, Object> unsetVals = values.containsKey("$unset") ?
+                (Map<String, Object>)values.remove("$unset") : new HashMap<>();
+        Map<String, Object> setVals = values.containsKey("$set") ?
+                (Map<String, Object>)values.remove("$set") : values;
+
+        return collectionFromStorageName(storageName, collection -> {
+            Document cleanQual = new Document((Map<String, Object>)cleanQualMap(qual));
+            Document updateVals = new Document();
+            if (!setVals.isEmpty()) {
+                if (setVals.containsKey("$rename")) {
+                    updateVals.put("$rename", setVals.get("$rename"));
+                } else {
+                    updateVals.put("$set", setVals);
+                }
+            }
+            if (!unsetVals.isEmpty()) {
+                updateVals.put("$unset", unsetVals);
+            }
+            UpdateOptions updateOptions = new UpdateOptions().upsert(false);
+            return collection.updateMany(cleanQual, updateVals, updateOptions);
+        }).singleElement().flatMap(updateResult -> {
+            if (updateResult.getModifiedCount() == 0) {
+                return Maybe.empty();
+            }
+            // not entirely sure what to return here... don't want to issue a find over the qual to get the updated
```

Comment:
Do we do OCC?  Could this be an OCC error?

Reply:
there's no OCC here (yet). we don't maintain an ars_version property for atlas data